src/http/request.cr

```diff
@@ -98,6 +98,19 @@ class HTTP::Request
     value
   end
 
+  # Return request host from headers
+  def host
+    host = @headers["Host"]?
+    return unless host
+    index = host.index(":")
+    index ? host[0...index] : host
```

Comment:
Speed benchmarks don't tell everything. What about memory usage? We merely want to return the Host header and want to be nice to the developer by extracting/skipping the `:port` part out of it. It's definitely not an URI, just a `host:port` definition.

Using `URI.parse` means that we instantiate an URI object on the Heap (it puts pressure on the GC) just to discard it immediately because we're only interested in the Host and maybe the Port values.

I believe reproducing [URI::Parse#parse_host](https://github.com/crystal-lang/crystal/blob/68783f1ebdd5b9bbefc8892ddbc200fe1f13bbea/src/uri/uri_parser.cr#L122) (which looks like what NGINX is doing) or the more [straightforward solution](https://github.com/crystal-lang/crystal/pull/3075#discussion_r73166742) from @RX14 would be more correct, and achieve the same.